spectrum-offchain-cardano/src/tx_tracker/pending_txs.rs

```diff
@@ -27,51 +28,62 @@ impl<TxHash, Tx> PendingTxs<TxHash, Tx> {
     {
         let should_confirm_until = self.current_block + self.max_confirmation_delay_blocks;
         self.index.insert(tx, should_confirm_until);
+        let now = Instant::now();
         match self.queue.entry(should_confirm_until) {
             Entry::Vacant(entry) => {
                 let mut txs = HashMap::new();
-                txs.insert(tx, trs);
+                txs.insert((tx, now), trs);
                 entry.insert(txs);
             }
             Entry::Occupied(mut entry) => {
-                entry.get_mut().insert(tx, trs);
+                entry.get_mut().insert((tx, now), trs);
             }
         }
     }
 
     pub fn confirm_tx(&mut self, tx: TxHash)
     where
-        TxHash: Eq + Hash + Display,
+        TxHash: Copy + Eq + Hash + Display,
     {
         if let Some(key) = self.index.remove(&tx) {
             if let Some(txs) = self.queue.get_mut(&key) {
-                let removed = txs.remove(&tx);
+                let instant = txs
+                    .keys()
+                    .find_map(|(tx_inner, instant)| if *tx_inner == tx { Some(instant) } else { None })
```

Comment:
@kettlebell you added timestamp into map keys and now have to do O(n) job to find a tx, why not storing ts in values?

Reply:
Yes, of course. Done.